include/odp/api/spec/pool.h

```diff
@@ -44,6 +44,78 @@ extern "C" {
 /** Maximum number of packet pool subparameters */
 #define ODP_POOL_MAX_SUBPARAMS  7
 
+/**
+ * Pool statistics counters options
+ *
+ * Pool statistics counters listed in a bit field structure.
+ */
+typedef union odp_pool_stats_opt_t {
+	/** Option flags */
+	struct {
+		/** @see odp_pool_stats_t::available */
+		uint64_t available          : 1;
+
+		/** @see odp_pool_stats_t::alloc_ops */
+		uint64_t alloc_ops          : 1;
+
+		/** @see odp_pool_stats_t::free_ops */
+		uint64_t free_ops           : 1;
+
+		/** @see odp_pool_stats_t::combined_ops */
+		uint64_t combined_ops       : 1;
+
+		/** @see odp_pool_stats_t::cache_available */
+		uint64_t cache_available    : 1;
+
+		/** @see odp_pool_stats_t::cache_alloc_ops */
+		uint64_t cache_alloc_ops    : 1;
+
+		/** @see odp_pool_stats_t::cache_free_ops */
+		uint64_t cache_free_ops     : 1;
+
+		/** @see odp_pool_stats_t::cache_combined_ops */
+		uint64_t cache_combined_ops : 1;
+	} bit;
+
+	/** All bits of the bit field structure
+	  *
+	  * This field can be used to set/clear all flags, or for bitwise
+	  * operations over the entire structure. */
+	uint64_t all;
+
+} odp_pool_stats_opt_t;
+
+/**
+ * Pool statistics counters
+ */
+typedef struct odp_pool_stats_t {
+	/** The number of available entries in the pool */
+	uint64_t available;
+
+	/** The number of alloc operations from the pool */
+	uint64_t alloc_ops;
```

Comment:
Alloc_ops and cache_alloc_ops counters should document if failed allocations (pool or cache is empty) are counted. May be those should not be counted. So, this would be "number of successful alloc operations from the pool". Counting only successful operations would show how often thread cache vs global pool returns events.